### DISCUSSION TITLE
Elasticsearch - cannot disciminate between multiple queries/frames with identical fields

### DISCUSSION THREAD

**Discussion by PayBas** [grafana/grafana#44400] [Data Sources]
I'm collecting build-data in Elasticsearch, and I'm trying to generate a single histogram panel which shows the build-time distributions for _multiple_ result categories. So a distribution of build-times for successful builds, a distribution of build-times for failed builds, etc.

So basically I want to transform this:
![image](https://user-images.githubusercontent.com/3015760/150846379-a52334fe-1388-43cc-a03a-5ff8a73dd463.png)

Into this:
![image](https://user-images.githubusercontent.com/3015760/150846036-b844e05f-4224-4484-835d-b30d7b2539d3.png)

Ultimately I would like to stack them, but that's a different issue (which hopefully will be addressed in https://github.com/grafana/grafana/issues/38047 ).

### The problem:
The problem in two-fold.

1. The "Histogram" panel only supports a single numerical data series per frame.
2. Splitting the data into multiple data frames leaves no method for discriminating between them when they have identical fields.

I'll try to explain as best I can.

|data.buildDuration|data.result.keyword|Count|
|---|---|---|
|8535800|FAILURE|1|
|7714800|SUCCESS|1|
|7480800|FAILURE|1|
|7437700|SUCCESS|1|
|7363000|FAILURE|1|
|7285600|SUCCESS|1|
|6532700|FAILURE|1|
|6458200|FAILURE|1|
|6174400|FAILURE|1|
|6169200|FAILURE|1|

The above data cannot be parsed correctly by the Histogram panel because the data is basically multiple series (based on `data.result`) squashed into one. There is no transformation available that I know of that can create a new series frame based on this input.

So I had to change the Elasticsearch query.
From:
![image](https://user-images.githubusercontent.com/3015760/150864052-75385fbf-9abb-4727-9a0d-3f6982675f33.png)
To:
![image](https://user-images.githubusercontent.com/3015760/150864128-b9647ad6-6607-4d25-a4ba-e271a009cc38.png)
And apply a `Goup by` transformation on the `data.buildDuration` field.

This gives for the `FAILURE` query:
|data.buildDuration|
|---|
|8535800|
|7480800|
|7363000|
|6532700|
|6458200|
|6174400|
|6169200|

And for the `SUCCESS` query:
|data.buildDuration|
|---|
|7714800|
|7437700|
|7285600|

Since these are 2 different queries, the results are available as 2 completely separate data frames (between which you can switch using the dropdown menu at the bottom of the table view). Neat!

This is great, because the Histogram panel handles them as 2 completely different series, which can be rendered independently.
However - and this is where my problem comes in - since the _fields_ of both queries are identical, there is no way to disciminate between the "SUCCESS" series and the "FAILURE" series in the panel options. They are _both_ `data.buildDuration`.

![image](https://user-images.githubusercontent.com/3015760/150865379-15bbab27-3bdd-457a-8a9f-335bf64c91e5.png)

So there is no way for me to control the formatting of the series individually.
I also cannot use the "Organize fields" transformation to rename fields, because I just split the query/data into 2 different frames (which the "Organize fields" transformation can't process).

Is there some way to discriminate the fields based on the query name, or some other method?

Any input is welcome.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
